### PR TITLE
Spanish translations to extend QALD-9-plus

### PR DESCRIPTION
**Pull request by javiersorucol**
Hi! 
We made Spanish translations manually by native speakers to extend QALD-9-plus. We modified:
- The files in the data folder, adding the new Spanish translations.
- The Readme.md file to include Spanish in the dataset description.
- The statistics.ipynb notebook, we ran it so it displays the updated statistics that includes Spanish, we had to modify  the function read_json to specify the encoding as utf-8 to make it work.

### CODE REVIEW

**Comment by anbo-de** [comment]
@javiersorucol Thank you very much for the effort you already put into these translations. We will take a look into the results. What kind of translation process did you use? Is there a description somewhere?

**Comment by javiersorucol** [comment]
Sorry for the late reply,
At the moment we are writing a paper about these translations, to perform these translations, the original QALD-9 Spanish translations were reviewed and corrected if necessary. I'll leave below the sections related to the QALD-9 translation errors and the translation process so you can have a full insight into the process.

## Qualitative Analysis of QALD-9 Spanish translations
QALD-9 dataset is composed of 558 questions, of which 408 correspond to the training dataset
and 150 to the testing dataset.
As we mentioned before, QALD-9 already has Spanish translations; after an initial review of
them, we came to the conclusion that, like other QALD-9 translations, the quality of the Spanish
ones is also doubtful.
We decided to review the original translations during the translation process, classifying each
translation into seven cases:
• The first case involves what we call ”minor translation mistakes,” which we define as
translation errors that do not alter the question’s original meaning. Some common errors
in this case are the use of the incorrect geners (for example, using the word ”hermoso”
for a female noun or subject), missing words that do not contribute to the meaning of the
question, the absence of opening and closing question marks (’¿?’), missing letters, the
wrong use of plural forms, the lack of capital letters in proper names, or using the wrong
tense.
An example of these issues can be found within the question ”Who developed Skype?”
which got translated to ”Quien desarrollado Skype?”; The verb ”desarrollado” in the past
participle form, to be a correct translation, the auxiliary verb ”ha” should accompany
the main verb resulting in the translation ”¿Quién ha desarrollado Skype?”. Another
correct alternative is to modify the main verb by changing it to the past tense, resulting
in ”¿Quién desarrolló Skype?”.
• The second case is about what we call ”major translation errors”, the main characteristic
of these errors is that they result in the loss of meaningful elements of the question, such
as verbs, proper names, and other meaningful words. One example of this case is found
in the question ”In which U.S. state is Area 51 located?” which got translated to ”En cual
Nosotros estado es Zona 51 ¿situado?”. If we reverse this translation, we would get ”In
which us state is Zone 51 located?” As you can see, the entity ”U.S.” was mistranslated
to the word ”us” and the entity ”Area 51” was transformed to ”Zone 51”. In both cases,
the entities in question were lost, meaning that a KGQA system would not be able to
work properly with this translation. Another example of this issue can be found in the
question ”Where did Abraham Lincoln die?” that gets translated to ”Dónde hizo Abrahán
Lincoln el?”. In this translation, the verb ”die” was completely lost, resulting in a question
that just does not make sense. This case is related to questions where the translation just
does not make sense. In the original QALD-9 translations, we found some questions that
lacked their corresponding Spanish translation. Those questions were classified in this
case. This case can be triggered by mistakes both in the question and in the question’s
keywords.
• The third case relates to questions where the question’s keywords require modifications
due to some mistakes in the question’s original translation that propagate to the question’s
keywords.
• The fourth case is similar to the third case, the difference is that in this case the errors
found in the question’s keywords are not related to errors in the question’s translation,
like words that are written properly in the question’s translation but are wrongly written
in the question’s keywords.
• The fifth case is about Spanish accentuation. The character ” ́” is known in Spanish as
”orthographic accent”. This accent is used on some words’ vowels and can modify the
word’s meaning. One example is the words ”mas” and ”más”. ”mas” is an equivalent of the
word ”but”, while ”más” is a quantity adverb. Some QALD-9 translations lack orthographic
accents, so the real meaning of some words is lost. This case can be triggered by mistakes
both in the question or in the question keywords.
• The sixth case is composed of questions with correct translations that need modifications
in order to be more natural. Modifications in the question’s keywords do not trigger this
case.
• Finally, the seventh case relates to questions that do not require any modification. Modi-
fications to the question’s keywords do not affect this case.

## Translation of Questions
The translation process consisted of two stages: translation and review.

In the translation stage, a native speaker called "the translator" reviewed each question by checking the original English question, its original QALD-9 translation, the English question keywords, and the QALD-9 translation keywords; then the translation annotated the cases that describe the question translation (see above) and generated a question translation for the English question. If the question's translation was found to be correct (case 7), the translation was kept. The translator will always look for the Spanish version of proper names in order to keep the translation as natural as possible (cases like "Iraq", which is spelled "Irak" in Spanish), this also implies not translating English proper names if the entity is known by the Spanish-speaking community by the English name (like the TV show "Friends"). The translator's goal is always to generate correct translations (avoid grammatical and accentuation mistakes) that fit in the natural Spanish dialect.

In the review stage, a second native speaker called "the reviewer" reviewed the original English question, the translation generated by the translator, and the keywords from the new translation, looking for possible mistakes. The mistakes found were annotated and sent to the translator.
The translator reviewed the observations received, and if the translator agreed with the correction, the question was modified; corrections that raised additional concerns about the translation were annotated and discussed with the reviewer. The reviewer exposed the motives for why the corrections are necessary, so the translator can make a choice; in some cases, the question's SPARQL query and answers were reviewed by the translator in order to make an appropriate choice.

This process was executed over the original QALD-9 dataset, once the native translations were generated, they were merged into QALD_9_plus by using the question's id, generating what we call QALD-9-ES.

**Comment by Perevalov** [comment]
@anbo-de Maybe, let's wait for the paper pre-print or actual publication and then merge it as well as mention the paper in the readme?
CC: @javiersorucol

**Comment by anbo-de** [comment]
I will merge the pull request to enable a resource paper submission. When the paper is accepted, we add the documentation/references, too.
